src/Microsoft.Health.Fhir.Core/Features/Operations/BulkDelete/Mediator/GetBulkDeleteHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.Health.Core.Features.Security.Authorization;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.BulkDelete.Mediator
+{
+    public class GetBulkDeleteHandler : IRequestHandler<GetBulkDeleteRequest, GetBulkDeleteResponse>
+    {
+        private IAuthorizationService<DataActions> _authorizationService;
+        private IQueueClient _queueClient;
+
+        public GetBulkDeleteHandler(
+            IAuthorizationService<DataActions> authorizationService,
+            IQueueClient queueClient)
+        {
+            _authorizationService = authorizationService;
+            _queueClient = queueClient;
+        }
+
+        public async Task<GetBulkDeleteResponse> Handle(GetBulkDeleteRequest request, CancellationToken cancellationToken)
+        {
+            if (await _authorizationService.CheckAccess(DataActions.Read, cancellationToken) != DataActions.Read)
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            var jobs = await _queueClient.GetJobByGroupIdAsync((byte)QueueType.BulkDelete, request.JobId, false, cancellationToken);
+            var failed = false;
+            var cancelled = false;
+            var succeeded = true;
+            var resourcesDeleted = new Dictionary<string, long>();
+            var issues = new List<OperationOutcomeIssue>();
+            var failureResultCode = HttpStatusCode.OK;
+
+            foreach (var job in jobs)
+            {
+                if (job.Status == JobStatus.Failed)
+                {
+                    failed = true;
+                    succeeded = false;
+
+                    // TODO: fix exception logging
```

Comment:
Should we fix the exception logging in this PR? Do you want to improve the "detailsText" with something more specific?

Reply:
Yes, this was a note to remind me to fix this in this PR.

Reply:
I'm fixing the error message to be something usable, but the error handing of async jobs needs to be reworked. Right now there is no way to deserialize the objects stored when error happen because the formatting isn't standardized. This will be a larger effort than I want to do as part of this story as it will involve reworking error handling for Import and Export as well.
I made a new story to track this work: [User Story 104493](https://microsofthealth.visualstudio.com/Health/_workitems/edit/104493): Rework async job error handling